### PR TITLE
Clean up after play kube failure

### PR DESCRIPTION
**Pull request by haircommander**
Before, we would half create a pod in play kube and error out if we fail.
Rather, let's clean up after our failure so the user doesn't have to delete the pod themselves.

Signed-off-by: Peter Hunt <pehunt@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
LGTM
/approve

File: cmd/podman/play_kube.go [hunk 1]
```diff
@@ -117,12 +118,25 @@ func playKubeYAMLCmd(c *cliconfig.KubePlayValues) error {
...
 	if err != nil {
 		return err
 	}
-	// Print the Pod's ID
-	fmt.Println(pod.ID())
+	// We've now created a resource that will need to be cleaned up if we fail.
+	defer func() {
+		if err != nil {
+			if containers != nil {
+				for _, ctr := range containers {
```
Comment (mheon) [inline cmd/podman/play_kube.go]: Actually, do we need to do this if we're already force-removing the pod? That ought to evict all remaining containers in it anyways.

Comment (haircommander) [reply cmd/podman/play_kube.go]: Ope yeah you're right

Comment (haircommander) [reply cmd/podman/play_kube.go]: Fixed

File: cmd/podman/play_kube.go [hunk 2]
```diff
@@ -117,12 +118,18 @@ func playKubeYAMLCmd(c *cliconfig.KubePlayValues) error {
...
-	pod, err := runtime.NewPod(ctx, podOptions...)
+	pod, err = runtime.NewPod(ctx, podOptions...)
 	if err != nil {
 		return err
 	}
-	// Print the Pod's ID
-	fmt.Println(pod.ID())
+	// We've now created a resource that will need to be cleaned up if we fail.
+	defer func() {
+		if err != nil {
```
Comment (vrothberg) [inline cmd/podman/play_kube.go]: The `err` in the loop starting in line 193 will be ignored as it's in a new scope (see minimal example https://play.golang.com/p/JMJT9T0Fbj6). Although I find the solution with deferred checks elegant, I think they are prone to errors.

I prefer to add a helper function and wrap around that with an error check. This way, we don't have to worry about future changes that potentially break the deferred check.

```golang
if err := helperFunc(...); err != nil {
   logrus.Debugf(err.Error())
   if errCleanUp := runtime.RemovePod(...); errCleanup != nil {
     logrus.Errorf(....)
  }
}

File: cmd/podman/play_kube.go [hunk 3]
```diff
@@ -206,6 +213,12 @@ func playKubeYAMLCmd(c *cliconfig.KubePlayValues) error {
 			// the other containers are in created status
 			return err
 		}
+	}
+
+	// We've now successfully converted this YAML into a pod
+	// print our pod and containers, signifying we succeeded
+	fmt.Println(pod.ID())
```
Comment (rhatdan) [inline cmd/podman/play_kube.go]: Should there be header here?  To allow users to differentiate between pod and containers?

File: cmd/podman/play_kube.go [hunk 4]
```diff
@@ -62,12 +62,13 @@ func init() {
 	flags.BoolVar(&playKubeCommand.TlsVerify, "tls-verify", true, "Require HTTPS and verify certificates when contacting registries")
 }
 
-func playKubeYAMLCmd(c *cliconfig.KubePlayValues) error {
+func playKubeYAMLCmd(c *cliconfig.KubePlayValues) (err error) {
```
Comment (rhatdan) [inline cmd/podman/play_kube.go]: Nit:  I prefer to capatalize this err.  (Err error), then you can continue to use the lower case err as you always would.

**Comment by mheon** [comment]
Restarted failing F29 test

**Comment by mheon** [comment]
One more comment to hit and I think we ought to be good

File: cmd/podman/play_kube.go (hunk 3 above)
Comment (haircommander) [reply cmd/podman/play_kube.go]: How does that look?

**Comment by haircommander** [comment]
@mheon happy tests and comments addressed PTAL

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**